internal/cli/clients/install_client.go

```diff
@@ -61,6 +61,8 @@ func (c *InstallClient) Install(cmd *cobra.Command) error {
 	defer log.Info("return")
 	details := log.V(1) // NOTE: Increment of level, not absolute.
 
+	ctx := context.Background()
```

Comment:
What is the purpose of passing a context everywhere, if that context has no way to be cancelled? As far as I understand a context exists as a way to signal various part of the code that what they are doing is no longer relevant (the went "out of context") and as such, they should stop doing it. This is the case with code that is triggered by http requests, when the request is no longer alive, or with code that it triggered from command line when the user does `ctrl+c`. Generally code that should run only as long as something else is still happening.

In this case, the only reason to cancel the context would be if the user stopped the installation (ctrl+c?) or if one of the deployments failed. We would need to create a "child" context with a CancelFunc in order to achieve this (as we do here for example: https://github.com/epinio/epinio/blob/main/internal/api/v1/applications.go#L236) and I don't see this here. Is this meant to happen in another PR?

Reply:
It's hard to implement a propagated context just in some part of the code base as the shared functions are changed to accept one.

I'm following convention here, by adding a context to all 'request' related funcs. Cobra also provides a top-level context  to commands, which we can propagate.

> A function that is never request-specific may use context.Background(), but err on the side of passing a Context even if you think you don't need to. The default case is to pass a Context; only use context.Background() directly if you have a good reason why the alternative is a mistake.
https://github.com/golang/go/wiki/CodeReviewComments#contexts

I like that this makes request-related funcs much more visible.

Improving cancel-ability we can do in another PR, as discussed.